src/Facade/Services/CircuitBoardFacadeService.cs

```diff
@@ -5,31 +5,100 @@
     using System.Linq;
     using System.Linq.Expressions;
 
+    using Linn.Common.Domain.Exceptions;
     using Linn.Common.Facade;
     using Linn.Common.Persistence;
+    using Linn.Common.Proxy.LinnApps;
     using Linn.Purchasing.Domain.LinnApps.Boms;
+    using Linn.Purchasing.Facade.Extensions;
     using Linn.Purchasing.Resources.Boms;
 
-    public class CircuitBoardFacadeService : FacadeResourceService<CircuitBoard, string, CircuitBoardResource, CircuitBoardResource>, ICircuitBoardFacadeService
+    public class CircuitBoardFacadeService :
+        FacadeResourceService<CircuitBoard, string, CircuitBoardResource, CircuitBoardComponentsUpdateResource>,
+        ICircuitBoardFacadeService
     {
+        private readonly ITransactionManager transactionManager;
+
+        private readonly IBuilder<CircuitBoard> resourceBuilder;
+
+        private readonly IRepository<PcasChange, int> pcasChangeRepository;
+
+        private readonly ICircuitBoardService circuitBoardService;
+
+        private readonly IDatabaseService databaseService;
+
         private readonly IQueryable<BoardRevisionType> types;
 
         public CircuitBoardFacadeService(
             IRepository<CircuitBoard, string> repository,
             ITransactionManager transactionManager,
             IBuilder<CircuitBoard> resourceBuilder,
-            IRepository<BoardRevisionType, string> boardRevisionTypeRepository)
+            IRepository<BoardRevisionType, string> boardRevisionTypeRepository,
+            IRepository<PcasChange, int> pcasChangeRepository,
+            ICircuitBoardService circuitBoardService,
+            IDatabaseService databaseService)
             : base(repository, transactionManager, resourceBuilder)
         {
+            this.transactionManager = transactionManager;
+            this.resourceBuilder = resourceBuilder;
+            this.pcasChangeRepository = pcasChangeRepository;
+            this.circuitBoardService = circuitBoardService;
+            this.databaseService = databaseService;
             this.types = boardRevisionTypeRepository.FindAll();
         }
 
         public IResult<CircuitBoardResource> UpdateBoardComponents(
-            int id,
-            CircuitBoardResource updateResource,
+            string id,
+            CircuitBoardComponentsUpdateResource updateResource,
             IEnumerable<string> privileges = null)
         {
-            throw new NotImplementedException();
+            var pcasChange = this.pcasChangeRepository.FindBy(
+                a => a.BoardCode == id && a.RevisionCode == updateResource.ChangeRequestRevisionCode
```

Comment:
Does this pcas_change need to be in any specific state? For example is it possible this FindBy could fish out a change that's already LIVE or CANCEL'd to track this update against?

Reply:
good point